Automattic/wp-calypso
# Improve data mocking approach in Storybook

Related to #89526

## Proposed Changes

Focuses on improving our approach to data mocking in Storybook. The key change is enabling API mocking, which allows us to avoid the need for overrides in Storybook (such as the `useGridPlans` hook). Overrides inevitably become a maintenance burden and increase the fragility of our stories. By mocking API responses and deferring to the intent system to control what/how plans are displayed, our Storybook environment is a closer reflection of production.

- Adds `msw-storybook-addon` to the `plans-grid-next` package to facilitate API mocking
- Fixtures which correspond to API responses have been added to `../src/fixtures`
    - `/plans`: `../src/fixtures/plans.ts`
    - `/sites/:id/plans`: `../src/fixtures/site-plans.ts`
    - `/sites/:id/purchases`: `../src/fixtures/sites-purchases.ts`
- Simplifies each `*.stories.*` file by moving any shared config/setup into `.storybook/preview.js`
- Lots of misc cleanup 

## Testing Instructions

1. Run storybook: `yarn workspace @automattic/plans-grid-next storybook`
2. View each story and verify that grids are appearing as expected
3. Open your browser console and filter by 'msw' to verify that API mocking is enabled and working:

![CleanShot 2024-05-02 at 14 18 19@2x](https://github.com/Automattic/wp-calypso/assets/69198925/de110cf0-4aff-4491-9fd9-ca8c419a557c)

4. Try changing the `intent` prop in the controls panel. The grid layout should update accordingly:

![CleanShot 2024-05-02 at 15 23 15](https://github.com/Automattic/wp-calypso/assets/69198925/40011357-a50e-4717-936e-258752b76a36)


## Pre-merge Checklist
